common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1123,7 +1123,9 @@ def _block_matches_all(block_data):
             settings['children'] = qualifiers.pop('children')
         for block_id, value in course.structure['blocks'].iteritems():
             if _block_matches_all(value):
-                items.append(block_id)
+                # Check if the block is in the course tree
+                if self.has_path_to_root(block_id, course):
```

Comment:
@adampalay I was just testing this on sandbox for cohort orphans issue.
Yes you are right, it should be behind a kwarg if that is argument is present then we should check if item is present in the course tree.